### PR TITLE
Update rank.go.tmpl

### PR DESCRIPTION
**Pull request by JordiHakimZiyech**
new PR, with updated code

### CODE REVIEW

File: leveling/rank.go.tmpl [hunk 1]
```diff
@@ -5,13 +5,19 @@
...
+{{/* CONFIGURATION AREA ENDS*/}}
 {{/* Instantiate constants */}}
 {{ $barEmpty := "□" }}
 {{ $barFull := "■" }}
 {{ $xp := 0 }} {{/* Xp of user */}}
 {{ $color := 14232643 }} {{/* Embed color */}} 
 {{ $user := .User }} {{/* Target user */}}
 {{ $colorSet := false }}
+{{ if (dbGet .User.ID "background") }}
+{{ $rankbackground = ((dbGet .User.ID "background").Value) }} {{end}}
```
Comment (Jo3-L) [inline leveling/rank.go.tmpl:20]: No way to set this currently, please remove or add a subcommand to set it (like how `set-color` is implemented).

File: leveling/rank.go.tmpl [hunk 2]
```diff
@@ -62,7 +68,7 @@
 	{{ range seq 1 11 }}
 		{{- if ge $percentage . }} {{- $bar = joinStr "" $bar $barFull }} {{/* We join with full bar unicode if percentage > current iteration */}}
 		{{- else }} {{- $bar = joinStr "" $bar $barEmpty }} {{- end -}} {{/* Otherwise, join with empty bar */}}
-	{{ end }}
+	{{- end -}}
```
Comment (Jo3-L) [inline leveling/rank.go.tmpl:71]: No need.
```suggestion
	{{ end }}
```

File: leveling/rank.go.tmpl [hunk 3]
```diff
@@ -73,9 +79,9 @@
 	}} {{/* Format embed */}}
 
 	{{ $rank := 0 }} {{/* Current rank of user */}}
-	{{ range $index, $entry := dbTopEntries "xp" 100 0 }} {{/* Loop over top 100 entries - see if user is in top 100 */}}
+	{{- range $index, $entry := dbTopEntries "xp" 100 0 -}} {{/* Loop over top 100 entries - see if user is in top 100 */}}
```
Comment (Jo3-L) [inline leveling/rank.go.tmpl:82]: No need.
```suggestion
	{{ range $index, $entry := dbTopEntries "xp" 100 0 }} {{- /* Loop over top 100 entries - see if user is in top 100 */}}
```

File: leveling/rank.go.tmpl [hunk 4]
```diff
@@ -73,9 +79,9 @@
...
 		{{- if eq .UserID $user.ID }} {{- $rank = add $index 1 }} {{- end -}}
-	{{ end }}
+	{{- end -}}
```
Comment (Jo3-L) [inline leveling/rank.go.tmpl:84]: No need.
```suggestion
	{{ end }}
```

File: leveling/rank.go.tmpl [hunk 5]
```diff
@@ -84,6 +90,11 @@
 			$embed.description
 		) }} {{/* If user in top 100, update the description */}}
 	{{ end }}
-
+{{ if $rankcard}}
+{{ if not $rank }} {{ $rank = urlescape "100" }} {{ end }}
```
Comment (Jo3-L) [inline leveling/rank.go.tmpl:94]: `100` is already url escaped.
Can also be onelined to this ->
```suggestion
{{ $rank = or $rank "100" }}
```

**Review by Jo3-L — CHANGES_REQUESTED** [review]

**Comment by JordiHakimZiyech** [comment]
me did an oopsie, i copied to wrong command version lmao, should be fine right now

**Review by Jo3-L — CHANGES_REQUESTED** [review]
1) is this tested?
2) some general questions

File: leveling/rank.go.tmpl [hunk 6]
```diff
@@ -26,14 +33,15 @@
 			{{ $hex := "D92C43" }}
 			{{ with index . 0 1 }}
 				{{ $hex = (printf "%06s" .) | upper }}
-			{{ end }}
+						{{ end }}
```
[the remark this answers is not in the document]
Comment (JordiHakimZiyech) [reply leveling/rank.go.tmpl:36]: my brain at 1am happened :thonk:

**Comment by Jo3-L** [comment]
lgtm, thanks!

**Pull request merged**